src/librustc_middle/mir/interpret/mod.rs

```diff
@@ -549,18 +549,20 @@ pub fn write_target_uint(
     mut target: &mut [u8],
     data: u128,
 ) -> Result<(), io::Error> {
-    let len = target.len();
     match endianness {
-        Endian::Little => target.write_uint128::<LittleEndian>(data, len),
-        Endian::Big => target.write_uint128::<BigEndian>(data, len),
-    }
+        Endian::Little => target.write(&data.to_le_bytes())?,
+        Endian::Big => target.write(&data.to_be_bytes())?,
+    };
+    Ok(())
 }
 
 #[inline]
 pub fn read_target_uint(endianness: Endian, mut source: &[u8]) -> Result<u128, io::Error> {
+    let mut buf = [0u8; std::mem::size_of::<u128>()];
+    source.read(&mut buf)?;
```

Comment:
And `read` also doesn't promise that the entire buffer is filled. Needs to use `read_exact` instead.

Reply:
My understanding based on testing with read_exact previously is that the buffer that read_target_uint is reading from is not actually guaranteed to be 16 bytes wide and that doing this returns UnexpectedEoF errors. I suppose I could handle that one case.